libdrgn/object.c

```diff
@@ -1075,7 +1075,7 @@ drgn_object_cast(struct drgn_object *res,
 		 struct drgn_qualified_type qualified_type,
 		 const struct drgn_object *obj)
 {
-	const struct drgn_language *lang = &drgn_language_c;
+	const struct drgn_language *lang = drgn_object_lang(obj);
```

Comment:
I believe we want `drgn_type_lang(qualified_type.type)` here, i.e., the language of the type we're casting to.